src/core_plugins/region_map/public/region_map_visualization.js

```diff
@@ -0,0 +1,137 @@
+import 'plugins/kbn_vislib_vis_types/controls/vislib_basic_options';
+import _ from 'lodash';
+import AggConfigResult from 'ui/vis/agg_config_result';
+import { BaseMapsVisualizationProvider } from '../../tile_map/public/base_maps_visualization';
+import ChoroplethLayer from './choropleth_layer';
+import { truncatedColorMaps }  from 'ui/vislib/components/color/truncated_colormaps';
+import AggResponsePointSeriesTooltipFormatterProvider from './tooltip_formatter';
+import 'ui/vis/map/service_settings';
+
+export function RegionMapsVisualizationProvider(Private, Notifier, config) {
+
+  const tooltipFormatter = Private(AggResponsePointSeriesTooltipFormatterProvider);
+  const BaseMapsVisualization = Private(BaseMapsVisualizationProvider);
+
+  return class RegionMapsVisualization extends BaseMapsVisualization {
+
+    constructor(container, vis) {
+      super(container, vis);
+      this._vis = this.vis;
+      this._choroplethLayer = null;
+      this._notify = new Notifier({ location: 'Region map' });
+    }
+
+
+    async render(esReponse, status) {
+      await super.render(esReponse, status);
+
+      if (!this._choroplethLayer) {
+        return;
+      }
+
+      if (this._choroplethLayer.isDataLoaded()) {
```

Comment:
Using a promise we would get rid of this if and the next return and just do something like `return this._choroplethLayer.whenDataLoaded()`.